biodatasets/msh_wsd/msh_wsd.py

```diff
@@ -0,0 +1,246 @@
+                gen_kwargs={
+                    "data_dir": Path(data_dir),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, data_dir: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        concepts = data_dir / "benchmark_mesh.txt"
+        with concepts.open() as f:
+            concepts = f.readlines()
+        concepts = [x.strip().split("\t") for x in concepts]
+
+        concept_map = {cuis[0]: {f"M{num}": cui for num, cui in enumerate(cuis[1:], 1)} for cuis in concepts}
+
+        files = list(data_dir.glob("*arff"))
+        for guid, file in enumerate(files):
+            if self.config.schema == "source":
+                for example in self._parse_document(concept_map, file):
+                    yield guid, example
+
+            elif self.config.schema == "bigbio_kb":
+                for document in self._parse_document(concept_map, file):
+                    for example in self._source_to_kb(document):
+                        yield example["id"], example
+
+    def _parse_document(self, concept_map, file: Path):
+
+        with file.open() as f:
+            content = f.readlines()
+        content = [x.strip() for x in content]
+
+        # search line number of @DATA, sometimes 6 or 7
+        start_l = None
+        for number, line in enumerate(content):
+            if line.startswith("@DATA"):
+                start_l = number + 1
+                break
+        assert start_l is not None
+
+        amb_word = file.with_suffix("").name[: -len("_pmids_tagged")]
+
+        sentences = []
+        for line in content[start_l:]:
+            # cant use , or ," ", as seperator
+            m_pmid = re.search("[0-9]+(?=(,))", line)
+            pmid = m_pmid.group()
+            m_label = re.search("(?<=(,))M[0-9]+", line)
+            label = m_label.group()
+
+            citation = line[m_pmid.span()[1] + 1 : m_label.span()[0] - 1].strip('"')
+
+            sentences.append({"pmid": pmid, "text": citation, "label": label})
+
+        yield {
+            "ambiguous_word": amb_word,
+            "sentences": sentences,
+            "choices": [{"label": key, "concept": value} for key, value in concept_map[amb_word].items()],
+        }
+
+    def _source_to_kb(self, document):
+        choices = {x["label"]: x["concept"] for x in document["choices"]}
+        for sentence in document["sentences"]:
+            document_ = {}
+            document_["events"] = []
+            document_["relations"] = []
+            document_["coreferences"] = []
+            document_["id"] = next(self.uid)
+            document_["document_id"] = sentence["pmid"]
+            document_["passages"] = [
+                {"id": next(self.uid), "type": "", "text": [sentence["text"]], "offsets": [[0, len(sentence["text"])]]}
+            ]
+            document_["entities"] = [
+                {
+                    "id": next(self.uid),
+                    "type": "",
```

Comment:
How about "ambigous_word"?

Reply:
Sorry, could you clarify? I don't understand what you mean unfortunately

Reply:
Just a minor thing `"type": "ambiguous_word"`

Reply:
Ah got you! Thanks!